### DISCUSSION TITLE
subscribeState andere Schreibweise

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2521] [General]
Was mache ich falsch?
Schreibweise funktioniert nicht

```
window.Socket.subscribeState(wifiDP)
.then((state) => {
	console.log(state.val);
})
.catch((err) => {
	console.error(`Subscribe State failed: ${err.message}`);
});
```
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/41994b10-6ae5-496a-8c5f-873f25fb29f7)

Normale Schreibweise funktioniert
```
window.Socket.subscribeState(wifiDP, ({ err, state }) => {
   /* Funktion die ausgeführt wird */
   console.log(state.val);
});
```

**Comment by Zefau** [comment]
`subscribeState` returned keinen Promise, sondern akzeptiert nur einen Callback, daher funktionieren `then` / `catch` nicht.

**Reply by mcuiobroker** [reply]
Ok. Danke.
